utils/build-script-impl

```diff
@@ -2265,17 +2440,17 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_COMPILE_TOOLS_DEPLOYMENT_TARG
                 if [[ -z "${INSTALL_LLDB}" ]] ; then
                     continue
                 fi
-                case "$(uname -s)" in
-                    Linux)
+                case ${host} in
+                    linux-*)
                         ;;
-                    FreeBSD)
+                    freebsd-*)
                         ;;
-                    CYGWIN_NT-10.0)
+                    cygwin-*)
                         ;;
-                    Darwin)
+                    macosx-*)
                         set_lldb_build_mode
                         pushd ${LLDB_SOURCE_DIR}
-                        xcodebuild -target toolchain -configuration ${LLDB_BUILD_MODE} install ${lldb_xcodebuild_options[@]} DSTROOT="${INSTALL_DESTDIR}" LLDB_TOOLCHAIN_PREFIX="${TOOLCHAIN_PREFIX}"
+                        xcodebuild -target toolchain -configuration ${LLDB_BUILD_MODE} install ${lldb_xcodebuild_options[@]} DSTROOT="${host_install_destdir}"
```

Comment:
When I tried it with the `LLDB_TOOLCHAIN_PREFIX` parameter, I got a strange output tree. In order to build a clean toolchain in both situations (cross-compiling and OSX-only), I had to:
- Omit the `LLDB_TOOLCHAIN_PREFIX` parameter. It seems to not always append a path separator, meaning I got output files in '${INSTALL_DESTDIR}${TOOLCHAIN_PREFIX}" - i.e "destdirswift3.xctoolchain"
- Include the `TOOLCHAIN_PREFIX` in the install location. Otherwise the swift compiler itself would install itself in to a `usr` folder adjacent to the toolchain location

I can't explain how it produced correct toolchains before. Both changes make sense to me. [This is a dump of the strange tree](https://github.com/apple/swift/files/281855/strange_tree.txt) I was getting without them.

.